src/preact/base-element.js

```diff
@@ -371,7 +414,22 @@ export class PreactBaseElement extends AMP.BaseElement {
     if (toHydrate) {
       hydrate(v, this.container_);
     } else {
-      render(v, this.container_);
+      const replacement = lightDomTag
+        ? childElementByTag(this.container_, lightDomTag)
+        : null;
+      if (replacement) {
+        replacement[RENDERED_PROP] = true;
+      }
+      render(v, this.container_, replacement);
+    }
+
+    // Dispatch the DOM_UPDATE event when rendered in the light DOM.
+    if (!isShadow) {
+      this.element.dispatchEvent(
```

Comment:
This is a bit dangerous to do, since `BaseElement` will be reused in the auto-envelope. This is a AMP-mode specific action, but will run in all modes.

Reply:
Understood, but we should probably move in a separate PR? It's been running at the same time just in a different place, so this code hasn't significantly changed the sequence of steps. Would you prefer to remove it for now? I don't think it's a usable code until we deal with the likes of `amp-list`.

Reply:
Also, is this even so bad to have in auto-envelope case? If so, we could also hide it behind the auto-envelope interface?